### DISCUSSION TITLE
Support for any ligand molecule

### DISCUSSION THREAD

**Discussion by demid-osipov** [jensengroup/propka#161] [Q&A]
Hello!
How do I make propka work with a ligand that is not among 18 ligands that are recognized?
Regards,
Demid Osipov

**Comment by sobolevnrm** [comment]
Can you provide an example of a ligand you're trying to use?

**Reply by demid-osipov** [reply]
There is no specific ligand, I am trying to come up with a general scheme for dealing with any unrecognized ligand. Shoud I somehow edit the source code?

**Reply by sobolevnrm** [reply]
I would start by looking at https://github.com/jensengroup/propka/blob/master/propka/ligand.py and https://github.com/jensengroup/propka/blob/master/propka/ligand_pka_values.py for ideas about how to generalize ligand support.

**Reply by demid-osipov** [reply]
Thank You.

**Reply by demid-osipov** [reply]
How can I determine that contribution to pKa from coulomb interaction between two groups depends on the total value of pKa of these groups?.. In other words, than adding a new ligand, how can a new line in interaction matrix be constructed?
